analysis/src/CompletionBackEnd.ml

```diff
@@ -1047,10 +1047,10 @@ let printConstructorArgs argsLen ~asSnippet =
   if List.length !args > 0 then "(" ^ (!args |> String.concat ", ") ^ ")"
   else ""
 
-type completionMode = Pattern | Expression
+type completionMode = Pattern of Completable.patternMode option | Expression
```

Comment:
I am wondering if we can get rid of the `option` here by having

```ocaml
type patternMode = Default | Destructuring
```

or something like that?

Reply:
Good point.